pkg/apis/scope/v0alpha1/types.go

```diff
@@ -100,3 +122,22 @@ type ScopeNodeList struct {
 
 	Items []ScopeNode `json:"items,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type TreeResults struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []TreeItem `json:"items,omitempty"`
+}
+
+type TreeItem struct {
+	NodeID   string   `json:"nodeId,omitempty"`
+	NodeType NodeType `json:"nodeType"` // container | leaf
+
+	Title       string `json:"title"`
+	Description string `json:"description,omitempty"`
+
+	LinkType LinkType `json:"linkType,omitempty"` // scope (later more things)
+	LinkID   string   `json:"linkID,omitempty"`   // the k8s name
```

Comment:
@kylebrandt @bergquist would it make sense to inline the `ScopeNodeSpec` here instead of duplicating the properties?
```suggestion
	ScopeNodeSpec `json:",inline"`
```

Reply:
I think that is fine - whatever your preference is.

Reply:
👍 unrelated e2e tests were acting flakey, so i merged this as is. it can be a future improvement in the next iteration if desired